### PR TITLE
Add shortcuts-search extension

### PR DESCRIPTION
**Pull request by solomkinmv**
## Description

Allows to list, search and run shortcuts for different applications.

There are 3 commands in total:

1. List all supported applications with shortcuts
2. List all shortcuts for active application
3. Copy application's bundle id to clipboard (useful for [adding new shortcuts](https://github.com/solomkinmv/shortcuts-disco/blob/main/README.md#shortcuts-contribution))

## Screencast

Screen Recording 2023-11-13 at 20 56 18

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

When checking an application that's not added, it just shows the empty view. Maybe you could tweak it to show an error?

shortcuts-search 2023-11-14 at 16 08 36


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/shortcuts-search/src/current-app.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Clipboard, getFrontmostApplication, showHUD } from "@raycast/api";
+
+export default async function Command() {
+  const frontmostApplication = await getFrontmostApplication();
+  if (frontmostApplication.bundleId) {
+    await Clipboard.copy(frontmostApplication.bundleId, { concealed: true });
```
[the remark this answers is not in the document]
Comment (solomkinmv) [reply extensions/shortcuts-search/src/current-app.ts:6]: No major reasons for concealed. I tried to keep clipboard history clean, but I guess that's redundant. Removed concealed property

[the remark this answers is not in the document]
Comment (solomkinmv) [reply extensions/shortcuts-search/src/current-app.ts]: Updated extension to close itself and show error HUD if no shortcuts available.
image

**Comment by solomkinmv** [comment]
Updated extension to close itself and show error HUD if no shortcuts available.
image

**Comment by pernielsentikaer** [comment]
There are a bit of [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) in the `All Shortcuts` command, can you look into this?

**Comment by solomkinmv** [comment]
> There is a bit of [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) in the `All Shortcuts` command, can you look into this?

Could you please check out latest version? I've fixed it in [this](https://github.com/raycast/extensions/pull/9235/commits/2e1bffe09df4b3d51efd5e173b9ab1fe5dc03c8e) commit. Double checked that it's still good after your improvements [here](https://github.com/raycast/extensions/pull/9235/commits/0a6aae6d52266adcd971efa5b36b446b99f5982f).

Here's slow motion recording of `List All Shortcuts` command after clearing caches and local storage.
Screen Recording 2023-11-15 at 21 39 25

**Comment by pernielsentikaer** [comment]
it's also when you go from `All shortcuts` and then click Open on an application, then you can see the flicker too - can you fix that too, otherwise let me know 🙂

**Comment by solomkinmv** [comment]
I've tried to fix that flickering "No results", but looks likethe problem is on the RayCast side.

To test this case I've created new extension with `useNavigation` and static list. This small  extension also has flickering "No results" and I can't control this. Even adding `isLoading={true}` doesn't prevent from flickering. 

Screen Recording 2023-11-18 at 00 50 57


```
import { Action, ActionPanel, List, useNavigation } from "@raycast/api";

export default function Command() {
  const { push } = useNavigation();
  return (
    <List>
      <List.Item
        key={"key"}
        title={"title"}
        actions={
          <ActionPanel>
            <Action title={"open"} onAction={() => push(<Command2 />)} />
          </ActionPanel>
        }
      /> </List>
  );
}

export function Command2() {
  return (
    <List>
      <List.Item
        key={"key"}
        title={"title2"}
      />
    </List>
  );
}
```

**Comment by pernielsentikaer** [comment]
You're right! We'll look into a fix 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**